Dockerfile

```diff
@@ -0,0 +1,32 @@
+# ---- Base Alpine with installed Node ----
+FROM alpine:3.14.3 AS base
+
+# install node
+RUN apk add --update nodejs npm nghttp2
+
+# ---- Install dependencies ----
+FROM base AS build
+
+WORKDIR /app
+COPY . .
+# install dependencies
+RUN npm ci
+
+# build to a production Javascript
+RUN npm run build:prod
+
+# ---- Serve ----
+FROM base AS release
```

Comment:
Isn't it the fact you are creating new images from `base` transferring also all files from that `base` image?
I think it is ok for intermediate layers but I guess you don't want all files in the final image and only `dist` directory?

Reply:
I need package*.json, openapi.yaml and dist. From Base I copy only npm and nodeJS, from build layer I copy dist but I also need only production dependencies :)

Reply:
I ran a quick test and It seems my concerns are real

<img width="554" alt="2022-01-05 at 12 46 21@2x" src="https://user-images.githubusercontent.com/1083296/148220282-cb419aab-49eb-4fbe-ae6f-5adbf73878d9.png">

Reply:
Maybe I don't understand, but if I have three images, base, build and release I copy all files from base to build and release even if release and build use that same. As I wrote I had a problem with openapi file. In `release` image I need to copy it and also `dist` folder from `build` with node and npm from `base` 😄

Reply:
Whenever you create a new image/layer based on another one, all the files are available in the new layer.

<img width="498" alt="2022-01-05 at 15 00 25@2x" src="https://user-images.githubusercontent.com/1083296/148239292-41d7c232-7252-42b3-b284-7e8d49410d07.png">

Reply:
Yes 😄  but we have two steps, build and release images based on `base` which is `alpine` + npm, node

Reply:
No problem 😃